diff -uNr a/src/compiler/spirv/meson.build b/src/compiler/spirv/meson.build
--- a/src/compiler/spirv/meson.build
+++ b/src/compiler/spirv/meson.build
@@ -87,10 +87,7 @@
      'vtn_bindgen2',
      ['vtn_bindgen2.c'],
      include_directories : [inc_include, inc_src],
      c_args : [c_msvc_compat_args, no_override_init_args],
      dependencies : [idep_vtn, idep_mesautil, idep_nir],
-     # If we can run host binaries directly, just build vtn_bindgen for the host.
-     # Most commonly this happens when doing a cross compile from an x86_64 build
-     # machine to an x86 host
-     native : not meson.can_run_host_binaries(),
+     native : false,
      install : get_option('install-mesa-clc'),
    )
 endif

